### DISCUSSION TITLE
Unable to add plugins

### DISCUSSION THREAD

**Discussion by M1gami** [Difegue/LANraragi#1141] [Q&A] [answered]
Most of my library comes from the nHentai Archivist(https://github.com/9-FS/nhentai_archivist) and it does not format the comicinfo.xml file the same way that the built in plugin is able to read. Previously, I was able to modify the existing plugin by locating the source for it, making the changes, and then overwriting the plugin with the modified version through LANraragi's plugin upload feature. This worked while I was using the previous version and I was able to run with this After updating to 0.9.22 I am no longer able to do this as it is resetting it to default. I did originally try to upload the plugin without overwriting the original but received the "could not load namespace" error below, and continued to receive it when attempted to upload it recently. Is there another workaround on this? I cannot seem to access the plugin folder directly.

**LRR Version and OS**
LRR 0.9.22, running on Windows 11, installed with the .msi from github, which is using WSL and Karen.exe

**Bug Details** 
the full error message that shows up, does not matter what I change the name of the file to, this always pops up. If I name it Comicinfo.pm it doesn't appear with the error but quickly resets it to the default plugin.
"Error uploading plugin
Could not load namespace LANraragi::Plugin::Metadata::ComicInfo - Copy! Your Plugin might not be compiling properly.
Here's an error log: "

**Comment by Difegue** [comment, marked as answer]
If you rename the plugin, you also need to adjust the namespace to match as it's linked to the filename - ie, `ComicInfo2.pm` should have a namespace of `LANraragi::Plugin::Metadata::ComicInfo2`. I would also avoid spaces in that filename.    

I'm not sure what would be causing sideloaded plugins to reset to default on WSL, there's zero integrity checks in place from my end at least so you should be able to mod the server folder at will..
